components/producers/producer_test.go

```diff
@@ -121,3 +123,29 @@ func TestParseMultiJSONMessages(t *testing.T) {
 	}
 	assert.Equal(t, want[0].Foo, "bar")
 }
+
+func TestGetPURLTarget(t *testing.T) {
+	target := GetPURLTarget("deb", "debian", "curl", "7.68.0", nil, "")
+	assert.Equal(t, "pkg:deb/debian/curl@7.68.0", target)
+
+	target = GetPURLTarget("bitbucket", "birkenfeld", "pygments-main", "244fd47e07d1014f0aed9c", nil, "")
+	assert.Equal(t, "pkg:bitbucket/birkenfeld/pygments-main@244fd47e07d1014f0aed9c", target)
+
+	target = GetPURLTarget("docker", "customer", "dockerimage", "sha256:244fd47e07d1004f0aed9c", packageurl.Qualifiers{
+		{Key: "repository_url", Value: "gcr.io"},
+	}, "")
+	assert.Equal(t, "pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io", target)
+
+	target = GetPURLTarget("npm", "", "foobar", "12.3.1", nil, "")
+	assert.Equal(t, "pkg:npm/foobar@12.3.1", target)
+
+	target = GetPURLTarget("pypi", "", "django", "1.11.1", nil, "")
+	assert.Equal(t, "pkg:pypi/django@1.11.1", target)
+
+	target = GetPURLTarget("deb", "debian", "curl", "7.50.3-1", packageurl.Qualifiers{
+		{Key: "arch", Value: "i386"},
+		{Key: "distro", Value: "jessie"},
+	}, "")
+
+	assert.Equal(t, "pkg:deb/debian/curl@7.50.3-1?arch=i386&distro=jessie", target)
+}
```

Comment:
can you also please add an test for the error case?
also, you don't need to test the third party library so much, i hope they have their own test cases :p 
you can do so with `require.Error()`

Reply:
actually there is no error, because of the types - it just passes everything down via the constructor, and then strings things together